### DISCUSSION TITLE
How does the `global` component parameter work?

### DISCUSSION THREAD

**Discussion by crotanite** [nuxt/nuxt#17881] [Questions]
Using a custom directory with `global: false` set within components doesn't seem to work for a dynamic component.

In the example below I setup 4 types:
1. The first is using the dynamic component to import directly from the `components` directory; this works.
2. The second uses an imported component from the custom directory as a dynamic component; this doesn't work.
3. The third uses the imported component from 2, but directly inserts the component itself; this works.
4. The fourth uses a non imported component from the custom directory, directly inserted; this works?? Should it?

The problem is 2 and 4. I had through that setting global to false would mean that I had to import the components to get them to work, but point 4 seems to invalidate this as without importing the components they still work. It also seems that even an imported component cannot be used with the dynamic component.

https://codesandbox.io/s/tender-sun-snvci?file=/app.vue

**Comment by danielroe** [comment]
With `global: false` components are imported wherever they are used, but not globally registered as async components (this is why #2 doesn't work and #4 works). You need `global: true` if you are using dynamic components.
